### PR TITLE
Construct MVPD partition in memory on startup

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
Didn't remove `mvpd_device_init()` and `mvpd_device_ro()` in case it will be needed.

### CODE REVIEW

File: src/soc/ibm/power9/mvpd.c [hunk 1]
```diff
@@ -16,13 +17,283 @@
...
+			/* This is a large (two-byte size) keyword */
+			if (read_eeprom(offset, &kwd_size, sizeof(kwd_size)) != sizeof(kwd_size))
+				die("Failed to read large keyword size from EEPROM\n");
+			kwd_size = le16toh(kwd_size);
+			offset += 2;
+		} else {
+			uint8_t small_size;
+			if (read_eeprom(offset, &small_size, sizeof(small_size)) != sizeof(small_size))
+				die("Failed to read small keyword size from EEPROM\n");
+			kwd_size = small_size;;
```
Comment (krystian-hebel) [inline src/soc/ibm/power9/mvpd.c]:
```suggestion
			kwd_size = small_size;
```

File: src/soc/ibm/power9/mvpd.c [hunk 2]
```diff
@@ -16,13 +17,283 @@
...
+			       const char *record_name, const char *kwd_name,
+			       uint8_t *buf, size_t *size)
+{
+	uint16_t record_size = 0;
+	uint8_t name[VPD_RECORD_NAME_LEN];
+
+	if (strlen(record_name) != VPD_RECORD_NAME_LEN)
+		die("Record name has wrong length: %s!\n", record_name);
+	if (strlen(kwd_name) != VPD_KWD_NAME_LEN)
+		die("Keyword name has wrong length: %s!\n", kwd_name);
```
Comment (krystian-hebel) [inline src/soc/ibm/power9/mvpd.c:137]: Is it always _exactly_ 4 or 2 bytes or can it be less?

File: src/soc/ibm/power9/mvpd.c [hunk 3]
```diff
@@ -16,13 +17,283 @@
...
+ *  </default>
+ * </attribute>
+ */
+
+/* Reads from a single EEPROM chip, which is deduced from offset. Returns number
+ * of bytes read. */
+static int read_eeprom_chip(uint32_t offset, void *data, uint16_t len)
+{
+	const unsigned int bus = 1;
+	uint16_t addr = 0xa0;
```
Comment (krystian-hebel) [inline src/soc/ibm/power9/mvpd.c]:
```suggestion
	uint16_t addr = 0xA0;
```

File: src/soc/ibm/power9/mvpd.c [hunk 4]
```diff
@@ -6,6 +6,7 @@
 #include <commonlib/region.h>
 #include <console/console.h>
 #include <cpu/power/vpd.h>
+#include <device/i2c_simple.h>
```
Comment (krystian-hebel) [inline src/soc/ibm/power9/mvpd.c:9]: This also requires adding `ramstage-y += i2c.c` to Makefile.inc

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: src/soc/ibm/power9/mvpd.c (hunk 2 above)
Comment (SergiiDmytruk) [reply src/soc/ibm/power9/mvpd.c:137]: AFAIK, yes, record and keyword names have fixed length.

**Review by krystian-hebel — APPROVED** [review]
Code for reading MVPD from PNOR stays for now. At some point we will have to test if it's worth to copy data to PNOR after reading, same as Hostboot does it.

**Pull request merged**